XenOvfApi/Security.cs

```diff
@@ -401,6 +392,52 @@ public static int CalculateStrength(string password)
         #endregion
 
         #region PRIVATE
+
+        private static Type GetAlgorithmClass(string key)
+        {
+            string algorithm = "";
+
+            if (!string.IsNullOrEmpty(key))
+            {
+                string[] parts = key.Split('#');
+
+                if (parts.Length > 1)
+                    algorithm = parts[1].ToLower().Replace('-', '_');
+            }
+
+            switch (algorithm)
+            {
+                case "base64":
+                    return typeof(FromBase64Transform);
+                case "rsa_1_5":
+                case "rsa_oaep_mgf1p":
+                    return typeof(RSACryptoServiceProvider);
+                case "tripledes_cbc":
+                case "kw_tripledes":
+                    return typeof(TripleDESCryptoServiceProvider);
+                case "sha1":
+                    return typeof(SHA1CryptoServiceProvider);
+                case "sha256":
+                    return typeof(SHA256CryptoServiceProvider);
+                case "sha384":
+                    return typeof(SHA384CryptoServiceProvider);
+                case "sha512":
+                    return typeof(SHA512CryptoServiceProvider);
+                case "des":
+                    return typeof(DESCryptoServiceProvider);
+                case "rc2":
+                    return typeof(RC2CryptoServiceProvider);
+                case "kw_aes128":
+                case "kw_aes256":
+                case "kw_aes192":
+                case "aes128_cbc":
+                case "aes256_cbc":
+                case "aes192_cbc":

```

Comment:
These are not used. I'd remove them unless they're here to aide future refactoring efforts.

Reply:
Indeed these can be removed because they fall into the default case, the only purpose they serve is to make explicit what algorithms one might expect. I think for the time being I'd ignore this because the list needs revisiting anyway (but it's not urgent since encryption is not supported yet).